typing-c/typing-c.cabal

```diff
@@ -0,0 +1,80 @@
+cabal-version:      3.0
+name:               typing-c
+version:            0.1.0.0
+license:            BSD-3-Clause
+author:             sheaf
+maintainer:         sheaf
+category:           System
+build-type:         Simple
+
+common common
+  ghc-options:
+    -Wall
+    -Wno-unticked-promoted-constructors
+  default-extensions:
+    DataKinds
+    DerivingStrategies
+    FlexibleInstances
+    GADTs
+    ImportQualifiedPost
+    LambdaCase
+    MagicHash
+    MultiParamTypeClasses
+    ParallelListComp
+    StandaloneKindSignatures
+    TupleSections
+    TypeApplications
+    TypeFamilies
+    TypeOperators
+
+  build-depends:
+      base
+        >= 4.16  && < 4.21
+    , template-haskell
+        >= 2.18  && < 2.23
+    , containers
+        >= 0.6   && < 0.8
+    , directory
+        >= 1.3   && < 1.4
+    , fin
+        >= 0.3.2 && < 0.4
+    , temporary
+        >= 1.2   && < 1.4
+    , text
+        >= 1.2   && < 2.2
+    , vec
+        >= 0.5   && < 0.6
+
+    , hs-bindgen-libclang
+    , hs-bindgen-patterns
+
+library
+  import:
+    common
+  hs-source-dirs:
+    src
+  exposed-modules:
+    C.Type
+    C.Typing
+    C.Typing.Arithmetic
+    C.Typing.Arithmetic.Classes
+    C.Typing.Arithmetic.Instances
+    C.Typing.Arithmetic.TH
+    C.Typing.Arithmetic.TyFams
+    C.Typing.Clang
```

Comment:
Separate out a simple "C.Expr" (platform-dependent) module for the C-like DSL, and then maybe a separate sublibrary for the stuff that hs-bindgen needs to import (or could be exposed `.Internal` modules)